templates/terraform/custom_delete/appversion_delete.go.erb

```diff
@@ -17,6 +17,16 @@ if err != nil {
 mutexKV.Lock(lockName)
 defer mutexKV.Unlock(lockName)
 
+// the default service cannot be deleted no matter what, so we'll return nil and log this warning instead
+if d.Get("service").(string) == "default" {
```

Comment:
This check means that no version in the default service can be deleted, right? The false case in the if block below. Even in the case it's the last version in the service, I think my preference would be to throw an error, requiring the user to use noop_on_destroy for the default service. I'd be afraid of a user with delete_service_on_destroy running a delete, getting a success, and being surprised that the version + service still exist.